src/meta/src/hummock/compaction/management/compactor_backend.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use risingwave_hummock_sdk::CompactionGroupId;
+use risingwave_pb::hummock::subscribe_compact_tasks_response::Task;
+use risingwave_pb::hummock::CompactTask;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+
+use super::CompactionTaskEvent;
+use crate::hummock::error::Error;
+use crate::hummock::HummockManagerRef;
+use crate::storage::MetaStore;
+
+#[async_trait]
+pub trait CompactorBackend {
+    // The `submit` interface accepts a compact task, and assigns the compact task to a remote for
+    // execution. The interface returns a unique id for the corresponding task, which is used for
+    // subsequent cancel.
+    async fn submit(
+        &self,
+        compaction_group: CompactionGroupId,
```

Comment:
Can we move the compaction_group to the task body? It looks like a parameter of the task.